src/bartiq/transform.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 PsiQuantum, Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import copy
+from typing import Any, Dict
+
+import sympy
+
+from bartiq import Resource, Routine
+from bartiq.compilation.types import Number
+from bartiq.symbolics import sympy_backend
+from bartiq.verification import verify_uncompiled_routine
+
+BACKEND = sympy_backend
+
+
+def add_aggregated_resources(aggregation_dict: Dict[str, Dict[str, Any]], routine: Routine, backend=BACKEND) -> Routine:
+    """
+    Add aggregated resources to bartiq routine based on the aggregation dictionary.
```

Comment:
As mentioned before, please add an example of how the `aggregation_dict` should look like in the docstring 🙏